File: daily/rust/713_subarray_product_less_than_k.rs
impl Solution {
    pub fn num_subarray_product_lt_k(nums: Vec<i32>, k: i32) -> i32 {
        if k == 0 {return 0;}
        let (mut cnt, mut pro) = (0, 1);
        let mut l  = 0;
        for r in 0..nums.len() {
            pro *= nums[r];
            while l <= r && pro >= k {
                pro /= nums[l];
                l+=1;
            }
            cnt += r - l + 1;
        }
        cnt as i32
    }
}

File: daily/cpp/713_subarray_product_less_than_k.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
 public:
  /**
   * @brief LC713: Subarray Product Less than K
   * Sliding window
   *
   * @param nums
   * @param k
   * @return int
   */
  int numSubarrayProdctLessThanK(vector<int>& nums, int k) {
    if (k == 0) return 0;
    int cnt = 0, pro = 1;
    for (int i = 0, j = 0; j < nums.size(); ++j) {
      pro *= nums[j];
      while (i <= j && pro >= k) {
        pro /= nums[i++];
      }
      cnt += j - i + 1;
    }
    return cnt;
  }
};